test/test_dynamic_shapes.py

```diff
@@ -19,6 +19,18 @@ def test_simple_expand(self):
     t5_cpu = t5.cpu()
     self.assertEqual(t5_cpu.shape[0], 1)
 
+  def test_simple_expand_on_2d_tensor(self):
+    size1 = 5
+    size2 = 2
+    t1 = torch.zeros([size1, size2], device=dev)
+    t1[3][0] = 1
+    t1[3][1] = 1
+    # t2 has size [<=10, 2]
+    t2 = torch.nonzero(t1)
+    t3 = torch.ones(1, size1, size2)
+    t4 = t3.expand(t2.shape[0], size1, size2)
```

Comment:
@ezyang Does pytorch supports one of the expand dimension is dynamic and others are static?